packages/api/__tests__/integration.test.js

```diff
@@ -0,0 +1,150 @@
+  //       'Content-Type': 'application/x-www-form-urlencoded',
+  //       'User-Agent': expect.stringMatching(/api \(node\)\/\d+.\d+.\d+/),
+  //     }),
+  //     json: null,
+  //     method: 'POST',
+  //     origin: expect.any(String),
+  //     url: 'https://httpbin.org/anything/v1/oa_citations/records',
+  //   });
+  // });
+});
+
+test('should support `image/png` requests', async () => {
+  const body = `${__dirname}/__fixtures__/owlbert.png`;
+
+  await expect(api(fileUploads).post('/anything/image-png', body)).resolves.toStrictEqual({
+    args: {},
+    data: expect.stringMatching(/data:application\/octet-stream;base64/),
+    files: {},
+    form: {},
+    headers: expect.objectContaining({
+      'Content-Type': 'image/png',
+      'User-Agent': expect.stringMatching(/api \(node\)\/\d+.\d+.\d+/),
+    }),
+    json: null,
+    method: 'POST',
+    origin: expect.any(String),
+    url: 'https://httpbin.org/anything/image-png',
+  });
+});
+
+describe('multipart/form-data', () => {
+  it('should support `multipart/form-data` requests', async () => {
+    const parametersStyle = require('@readme/oas-examples/3.1/json/parameters-style.json');
+
+    const body = {
+      primitive: 'string',
+      array: ['string'],
+      object: {
+        foo: 'foo-string',
+        bar: 'bar-string',
+      },
+    };
+
+    await expect(api(parametersStyle).post('/anything/form-data/form', body)).resolves.toStrictEqual({
+      args: {},
+      data: '',
+      files: {},
+      form: {
+        array: 'string',
+        object: 'foo,foo-string,bar,bar-string',
+        primitive: 'string',
+      },
+      headers: expect.objectContaining({
+        'Content-Type': expect.stringMatching(/^multipart\/form-data; boundary=form-data-boundary-(.*)$/),
+        'User-Agent': expect.stringMatching(/api \(node\)\/\d+.\d+.\d+/),
+      }),
+      json: null,
+      method: 'POST',
+      origin: expect.any(String),
+      url: 'https://httpbin.org/anything/form-data/form',
+    });
+  });
+
+  it('should support `multipart/form-data` requests with files', async () => {
+    const body = {
+      orderId: 1234,
+      userId: 5678,
+      documentFile: `${__dirname}/__fixtures__/hello.txt`,
+    };
+
+    await expect(api(fileUploads).post('/anything/multipart-formdata', body)).resolves.toStrictEqual({
+      args: {},
+      data: '',
+      files: {
+        documentFile: `Hello world!
+`,
+      },
+      form: { orderId: '1234', userId: '5678' },
+      headers: expect.objectContaining({
+        'Content-Type': expect.stringMatching(/^multipart\/form-data; boundary=form-data-boundary-(.*)$/),
+        'User-Agent': expect.stringMatching(/api \(node\)\/\d+.\d+.\d+/),
+      }),
+      json: null,
+      method: 'POST',
+      origin: expect.any(String),
+      url: 'https://httpbin.org/anything/multipart-formdata',
+    });
+  });
+
+  // it.only('should support `multipart/form-data` requests with an array of files', async () => {
```

Comment:
This doesn't send anything right now and is a larger problem with the `prepareParams` library that I plan on resolving in another PR.